Purchases/Attribution/RCAttributionFetcher.m

```diff
@@ -137,15 +137,32 @@ - (BOOL)isAuthorizedToPostSearchAds {
 
         BOOL needsTrackingAuthorization = [self.systemInfo isOperatingSystemAtLeastVersion:minimumOSVersionRequiringAuthorization];
 
-        Class<FakeATTrackingManager> _Nullable trackingManagerClass = [self.attributionFactory atTrackingManagerClass];
-        if (!trackingManagerClass && needsTrackingAuthorization) {
-            RCWarnLog(@"%@", RCStrings.attribution.search_ads_attribution_cancelled_missing_att_framework);
+        Class _Nullable trackingManagerClass = [self.attributionFactory atTrackingClass];
+        if (!trackingManagerClass) {
+            if (needsTrackingAuthorization) {
+                RCWarnLog(@"%@", RCStrings.attribution.search_ads_attribution_cancelled_missing_att_framework);
+            }
+            return !needsTrackingAuthorization;
+        }
+        SEL authStatusSelector = NSSelectorFromString(self.attributionFactory.authorizationStatusPropertyName);
+        BOOL canPerformSelector = [trackingManagerClass respondsToSelector:authStatusSelector];
+        if (!canPerformSelector) {
+            RCWarnLog(@"%@", RCStrings.attribution.att_framework_present_but_couldnt_call_tracking_authorization_status);
             return NO;
         }
-        NSInteger authorizationStatus = [trackingManagerClass trackingAuthorizationStatus];
-        BOOL authorized = authorizationStatus == FakeATTrackingManagerAuthorizationStatusAuthorized
+
+        NSMethodSignature *methodSignature = [trackingManagerClass methodSignatureForSelector:authStatusSelector];
+        NSInvocation *myInvocation = [NSInvocation invocationWithMethodSignature:methodSignature];
+        [myInvocation setTarget:trackingManagerClass];
+        [myInvocation setSelector:authStatusSelector];
+
+        [myInvocation invoke];
+        NSInteger authorizationStatus;
+        [myInvocation getReturnValue:&authorizationStatus];
```

Comment:
this is the very cumbersome equivalent of just doing `[trackingManagerClass performSelector:authStatusSelector]`, but at least it won't cause leaks